apis/rust/node/src/event_stream/thread.rs

```diff
@@ -230,12 +237,13 @@ fn report_remaining_drop_tokens(
                     drop_tokens.push(token);
                 }
                 Err(flume::RecvTimeoutError::Timeout) => {
-                    let duration = Duration::from_secs(30);
+                    let duration = Duration::from_secs(2);
                     if since.elapsed() > duration {
                         tracing::warn!(
                             "timeout: node finished, but token {token:?} was still not \
                             dropped after {duration:?} -> ignoring it"
                         );
+                        drop_tokens.push(token);
```

Comment:
Unfortunately this operation is a bit dangerous. Once the drop token is freed, the sender is allowed to overwrite the shared memory that backs the event. Using the event in any way after this can quickly lead to undefined behavior. So I'm a bit hesitant on this change.

Reply:
So, currently, what is happening is that no sending node can exit even though the node finished, which is very confusing.

I think that using an event after a node finishes is out of our scope for us, so I would prefer to drop the shared memory instead of making the dataflow eternally stuck as it is currently the case.

Reply:
This was replaced with dropping the drop token on the sender side leaving the shared memory region open as long as the memory is mapped.